tests_e2e/tests/scripts/agent_firewall-verify_all_firewall_rules.py

```diff
@@ -0,0 +1,427 @@
+    """
+    This function is used to verify all the iptable rules are present in the rule set
+    """
+    log.info("-----Verifying all ip table rules are present in rule set")
+    retry: int = 0
+    found: bool = False
+    while retry < max_retry and not found:
+        root_accept, non_root_accept, non_root_drop = get_all_iptable_rule_commands("-C")
+        found = check_if_iptable_rule_is_available(root_accept) and check_if_iptable_rule_is_available(
+            non_root_accept) and check_if_iptable_rule_is_available(non_root_drop)
+
+        if not found:
+            log.info("Not all ip table rules are present, retrying in 30 secs.\n")
+            time.sleep(30)
+        retry += 1
+
+    if not found:
+        fail("IP table rules missing in rule set.\n Current iptable rules:\n {0}".format(
+            print_current_iptable_rules()))
+
+    log.info("verified All ip table rules are present in rule set")
+
+
+def verify_rules_deleted_successfully(commands: List[List[str]] = None) -> None:
+    """
+    This function is used to verify if provided rule or all(if not specified) iptable rules are deleted successfully.
+    """
+    log.info("-----Verifying requested rules deleted successfully")
+
+    if commands is None:
+        commands = []
+
+    if not commands:
+        root_accept, non_root_accept, non_root_drop = get_all_iptable_rule_commands("-C")
+        commands.extend([root_accept, non_root_accept, non_root_drop])
+
+    # "-C" return error code 1 when not available which is expected after deletion
+    for command in commands:
+        if not check_if_iptable_rule_is_available(command):
+            pass
+        else:
+            raise Exception("Deletion of ip table rules not successful\n.Current ip table rules:\n" + print_current_iptable_rules())
+
+    log.info("ip table rules deleted successfully \n %s", commands)
+
+
+def delete_iptable_rules(commands: List[List[str]] = None) -> None:
+    """
+    This function is used to delete the provided rule or all(if not specified) iptable rules
+    """
+    if commands is None:
+        commands = []
+    if not commands:
+        root_accept, non_root_accept, non_root_drop = get_all_iptable_rule_commands("-C")
+        commands.extend([root_accept, non_root_accept, non_root_drop])
+
+    log.info("-----Deleting ip table rules \n %s", commands)
+
+    try:
+        for command in commands:
+            execute_cmd(command)
+    except Exception as e:
+        raise Exception("Error -- Failed to Delete the ip table rule set {0}".format(e))
+
+    log.info("Success --Deletion of ip table rule")
+
+
+def verify_dns_tcp_to_wireserver_is_allowed(user: str) -> None:
+    """
+    This function is used to verify if tcp to wireserver is allowed for the given user
+    """
+    log.info("-----Verifying DNS tcp to wireserver is allowed")
+    switch_user(user)
+    try:
+        socket.create_connection((get_wireserver_ip(), 53), timeout=10)
```

Comment:
we should probably increase the timeout (say 30s) to help rule out other reasons for the timeout